src/Features/Core/Portable/NavigateTo/NavigateToSearcher.cs

```diff
@@ -121,7 +121,7 @@ private async Task SearchAllProjectsAsync(bool isFullyLoaded)
             {
                 // make sure we only process this project if we didn't already process it above.
                 if (processedProjects.Add(currentProject))
-                    tasks.Add(Task.Run(() => SearchAsync(currentProject, priorityDocs.ToImmutableArray(), seenItems, isFullyLoaded), _cancellationToken));
+                    tasks.Add(Task.Run(() => SearchAsync(currentProject, priorityDocs.Where(d => d.Project == currentProject).ToImmutableArray(), seenItems, isFullyLoaded), _cancellationToken));
```

Comment:
previously we could search a project, but then say: hey... tehse *docs* from this *other* project should be prioritized.  It's pretty non-sensical and led to more complex reasoning downsteram.  Now, i've strictly doc'ed in teh interface that the priority docs are always from the project being passed in, and it means cleaner code and easier reasoning downstream.